astropy/cosmology/core.py

```diff
@@ -350,9 +356,8 @@ def __equiv__(self, other: Any, /) -> bool:
 
         # Check all parameters in 'other' match those in 'self' and 'other' has
         # no extra parameters (latter part should never happen b/c same class)
-        return set(self.__all_parameters__) == set(other.__all_parameters__) and all(
-            np.all(getattr(self, k) == getattr(other, k))
-            for k in self.__all_parameters__
+        return self._parameters_all == other._parameters_all and all(
```

Comment:
Could one not just do `self._parameters == other._parameters`? That checks values too, and derived ones presumably are the same (or could be tested too.

Reply:
SGTM.

Reply:
Unfortunately this won't work because `m_nu` is an array and ``self.parameters == other.parameters`` will fail.

Reply:
Ah, I missed that. It is something I have thought on and off about: whether `bool(bool_array)` should not just be equivalent to `bool_array.all()`. But it would be tricky: for `bool(a != b)`, one might want `.any()`... So, perhaps one should continue to refuse the temptation to guess.

Reply:
SGTM. I'll resolve this thread. 
I would love if dataclasses supported `attrs` eq customization (see the bottom of https://www.attrs.org/en/stable/comparison.html#customization for a numpy example)

Reply:
Interesting! Though it does become a bit of a mini-language...